Pear/Controller/ChatViewController.swift

```diff
@@ -0,0 +1,331 @@
+        ]
+        backButton.setImage(UIImage(named: "backArrow"), for: .normal)
+        backButton.addTarget(self, action: #selector(backPressed), for: .touchUpInside)
+        navigationItem.leftBarButtonItem = UIBarButtonItem(customView: backButton)
+
+        chatTableView.separatorStyle = .none
+        chatTableView.backgroundColor = .backgroundLightGreen
+        chatTableView.register(ChatTableViewCell.self, forCellReuseIdentifier: ChatTableViewCell.reuseId)
+        chatTableView.dataSource = self
+        chatTableView.delegate = self
+        view.addSubview(chatTableView)
+
+        setupChatInput()
+        getChatMessages()
+        setupConstraints(keyboardHeight: 0)
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        navigationController?.interactivePopGestureRecognizer?.delegate = self
+    }
+
+    @objc func backPressed() {
+        navigationController?.popViewController(animated: true)
+    }
+
+    private func hideKeyboardWhenViewTapped() {
+        let tap = UITapGestureRecognizer(target: self, action: #selector(dismissKeyboard))
+        tap.cancelsTouchesInView = false
+        tap.delegate = self
+        view.addGestureRecognizer(tap)
+    }
+
+    @objc private func dismissKeyboard() {
+        view.endEditing(true)
+    }
+
+    private func setupChatInput() {
+        chatInputContainerView.backgroundColor = .backgroundLightGreen
+        view.addSubview(chatInputContainerView)
+
+        chatInputTextField.attributedPlaceholder = NSAttributedString(string: "Say hi", attributes: [NSAttributedString.Key.foregroundColor: UIColor.greenGray, NSAttributedString.Key.font: UIFont.getFont(.book, size: 16)])
+        chatInputTextField.layer.cornerRadius = 12
+        chatInputTextField.backgroundColor = .white
+        chatInputTextField.delegate = self
+        chatInputContainerView.addSubview(chatInputTextField)
+
+        let leftPaddingView = UIView(frame: CGRect(x: 0, y: 0, width: 10, height: 0))
+        chatInputTextField.leftView = leftPaddingView
+        chatInputTextField.leftViewMode = .always
+
+        let rightPaddingView = UIView(frame: CGRect(x: 0, y: 0, width: 10, height: 0))
+        chatInputTextField.rightView = rightPaddingView
+        chatInputTextField.rightViewMode = .always
+
+        sendMessageButton.setImage(UIImage(named: "sendMessage"), for: .normal)
+        sendMessageButton.addTarget(self, action: #selector(sendMessage), for: .touchUpInside)
+        chatInputContainerView.addSubview(sendMessageButton)
+
+        separatorView.backgroundColor = .inactiveGreen
+        chatInputContainerView.addSubview(separatorView)
+    }
+
+    private func getChatMessages() {
+        databaseRef.child("User-Messages").child(currentUser.netID).child(messageUser.netID).observe(.childAdded) { snapshot in
+            let messageId = snapshot.key
+            let messageRef = self.databaseRef.child("Messages").child(messageId)
```

Comment:
maybe put these database strings into `Constants.swift`

Reply:
it says self is required because you're in the closure